engines/director/lingo/xlibs/spacemgr.cpp

```diff
@@ -0,0 +1,670 @@
+			curNode = "";
+			curView = "";
+			SpaceCollection &sc = me->_spaceCollections.getVal(curSpaceCollection);
+			if (!(sc.spaces.contains(curSpaceCollection) && me->_checkForDups)) {
+				sc.spaces[curSpace] = Space();
+			}
+		} else if ((int)result.find("NODE ", pointer) == pointer) {
+			pointer += 5;
+			while (pointer < (int)index && result[pointer] == ' ')
+				pointer++;
+			curNode = result.substr(pointer, index - pointer);
+			curView = "";
+			SpaceCollection &sc = me->_spaceCollections.getVal(curSpaceCollection);
+			Space &s = sc.spaces.getVal(curSpace);
+			if (!(s.nodes.contains(curNode) && me->_checkForDups)) {
+				s.nodes[curNode] = Node();
+			}
+		} else if ((int)result.find("VIEW ", pointer) == pointer) {
+			pointer += 5;
+			while (pointer < (int)index && result[pointer] == ' ')
+				pointer++;
+			int idEnd = pointer;
+			while (idEnd < (int)index && result[idEnd] != ' ')
+				idEnd++;
+			curView = result.substr(pointer, idEnd - pointer);
+			pointer = idEnd;
+			while (pointer < (int)index && result[pointer] != ' ')
+				pointer++;
+			SpaceCollection &sc = me->_spaceCollections.getVal(curSpaceCollection);
+			Space &s = sc.spaces.getVal(curSpace);
+			Node &n = s.nodes.getVal(curNode);
+			if (!(n.views.contains(curView) && me->_checkForDups)) {
+				n.views[curView] = result.substr(pointer, index - pointer);
+			}
+		} else {
+			warning("SpaceMgr::m_parseText: Unhandled instruction %s", result.substr(lastIndex, index).c_str());
+		}
+
+		lastIndex = (int)index + 1;
+	}
+	if (debugLevelSet(5)) {
+		Common::String format = result;
+		for (int i = 0; i < (int)format.size(); i++) {
+			if (format[i] == '\r')
+				format.replace(i, 1, "\n");
+		}
+		debugC(5, kDebugXObj, "SpaceMgr::m_parseText(): %s", format.c_str());
+	}
+
+	g_lingo->push(Datum(0));
+}
+
+void SpaceMgr::m_getCurData(int nargs) {
+	if (nargs != 0) {
+		warning("SpaceMgr::m_getCurData: expected 0 arguments");
+		g_lingo->dropStack(nargs);
+	}
+	SpaceMgrXObject *me = static_cast<SpaceMgrXObject *>(g_lingo->_currentMe.u.obj);
+	Common::String result;
+	if (me->_spaceCollections.contains(me->_curSpaceCollection)) {
+		result += Common::String::format("SPACECOLLECTION %s", me->_curSpaceCollection.c_str());
+		SpaceCollection &sc = me->_spaceCollections.getVal(me->_curSpaceCollection);
+		if (sc.spaces.contains(me->_curSpace)) {
+			result += "\r";
+			result += Common::String::format("SPACE %s", me->_curSpace.c_str());
+			Space &s = sc.spaces.getVal(me->_curSpace);
+			if (s.nodes.contains(me->_curNode)) {
+				result += "\r";
+				result += Common::String::format("NODE %s", me->_curNode.c_str());
+				Node &n = s.nodes.getVal(me->_curNode);
+				if (n.views.contains(me->_curView)) {
+					result += "\r";
+					result += Common::String::format("VIEW %s", me->_curView.c_str());
+				}
+			}
+		}
+	}
+
+	if (debugLevelSet(5)) {
+		Common::String format = result;
+		for (int i = 0; i < (int)format.size(); i++) {
```

Comment:
Why not use `uint` here? This way you avoid cast in the loop condition, and after all `String::replace` takes `uint` as its first argument.

Reply:
Fixed